yolox/tracker/byte_tracker.py

```diff
@@ -143,20 +150,29 @@ def __repr__(self):
 
 
 class BYTETracker(object):
-    def __init__(self, args, frame_rate=30):
+    def __init__(self, args, feature_extractor, global_stracks:List[STrack]=[], frame_rate=30):###<----- added feature_extractor & global_stracks attribute to constructor
         self.tracked_stracks = []  # type: list[STrack]
         self.lost_stracks = []  # type: list[STrack]
-        self.removed_stracks = []  # type: list[STrack]
 
+        ###-------------------edit-----------------------------###
+        self.global_stracks = global_stracks
+        self.removed_stracks = [STrack(track.tlwh, track.score) for track in global_stracks]  # type: list[STrack]
+        for r_track,g_track in zip(self.removed_stracks,global_stracks):
+            r_track.track_id = g_track.track_id
+            r_track.feature = g_track.feature
+        ###-------------------edit-----------------------------###
         self.frame_id = 0
         self.args = args
         #self.det_thresh = args.track_thresh
         self.det_thresh = args.track_thresh + 0.1
         self.buffer_size = int(frame_rate / 30.0 * args.track_buffer)
         self.max_time_lost = self.buffer_size
         self.kalman_filter = KalmanFilter()
+        ###------------edit------------###
+        self.extractor = feature_extractor
+        ###------------edit------------###
 
-    def update(self, output_results, img_info, img_size):
+    def update(self, frame, output_results, img_info, img_size): ###<----- added frame attribute to update method
```

Comment:
it would be fine to me, but maybe need documented a bit? might be confused to others?

Reply:
yeah, you are right. I have used the README.md file from the original ByteTracker repo as is, and that could be confusing. I will modify it to specify what changes I have made and how it is different from the original repository.

Reply:
This code snippet utilizes a modified update() method that takes a frame as an argument to extract features for each individual detected in the image. Here's a breakdown of the process:

Step 1: Person Detection

The code uses YOLOv8 (or any other object detector) to detect individuals in the frame.
YOLOv8 returns a result containing bounding boxes, class IDs, confidence scores, and other information for each detected object.
Step 2: Tracking with BYTETracker

The detected bounding boxes are passed to the BYTETracker class along with the frame.
BYTETracker is responsible for tracking individual objects across multiple frames.
Step 3: Region Cropping

The code crops the regions specified by the bounding boxes from the frame.
These cropped regions represent the individual persons detected in the image.
Step 4: Feature Extraction

The cropped regions are passed as a list to a feature_extractor function.
This function extracts features from each individual region, such as facial features or body posture.
Reason for Taking frame as Input:

The modified update() method takes the frame as input to enable feature extraction for each individual detected in the image. This allows for more accurate tracking and identification of individuals across multiple frames.

Reply:
Love the changes! I was working on ByteTrack with DETR and Yolov8 before, even with ReID model, the results seem still bad as I want to get more accurate results for unique counts. Hope this improve the results!